File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        res = 0
        for i in range(len(arr)):
            cur_sum = 0
            for j in range(i, len(arr)):
                cur_sum += arr[j]
                res += cur_sum if (j - i + 1) % 2 != 0 else 0
        return res

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.kt
class Solution {
    fun sumOddLengthSubarrays(arr: IntArray): Int {
        var res = 0
        for (i in 0..(arr.size-1)) {
            var cur_sum = 0
            for (j in i..(arr.size-1)) {
                cur_sum += arr[j]
                res += if ((j - i + 1) % 2 != 0) cur_sum else 0
            }
        }
        return res
    }
}